### PR TITLE
Test checkpoint variable substitution for JDBC DS on restore

### PR DESCRIPTION
**Pull request by tjwatson**
With a parameterized server.xml only set the environment values after checkpoint.  On restore varify the DS is updated properly.

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

File: dev/io.openliberty.data.internal_fat_jpa/fat/src/test/jakarta/data/jpa/DataJPATestCheckpoint.java [hunk 1]
```diff
@@ -77,13 +74,4 @@ public static void setUp() throws Exception {
     public static void tearDown() throws Exception {
         server.stopServer();
     }
-
-    static void configureEnvVariable(LibertyServer server, Map<String, String> newEnv) throws Exception {
-        Properties serverEnvProperties = new Properties();
-        serverEnvProperties.putAll(newEnv);
-        File serverEnvFile = new File(server.getFileFromLibertyServerRoot("server.env").getAbsolutePath());
```
Comment (KyleAure) [inline dev/io.openliberty.data.internal_fat_jpa/fat/src/test/jakarta/data/jpa/DataJPATestCheckpoint.java:84]: I think I put a blank `server.env` in server root to accomplish this.  Can you remove that file in this PR?

File: dev/io.openliberty.data.internal_fat/fat/src/test/jakarta/data/DataTestCheckpoint.java [hunk 2]
```diff
@@ -93,13 +92,4 @@ public static void setUp() throws Exception {
     public static void tearDown() throws Exception {
         server.stopServer();
     }
-
-    static void configureEnvVariable(LibertyServer server, Map<String, String> newEnv) throws Exception {
-        Properties serverEnvProperties = new Properties();
-        serverEnvProperties.putAll(newEnv);
-        File serverEnvFile = new File(server.getFileFromLibertyServerRoot("server.env").getAbsolutePath());
```
Comment (KyleAure) [inline dev/io.openliberty.data.internal_fat/fat/src/test/jakarta/data/DataTestCheckpoint.java:100]: I think I put a blank server.env in server root to accomplish this. Can you remove that file in this PR?

**Review by KyleAure — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.data.internal_fat_jpa/fat/src/test/jakarta/data/jpa/DataJPATestCheckpoint.java (hunk 1 above)
Comment (tjwatson) [reply dev/io.openliberty.data.internal_fat_jpa/fat/src/test/jakarta/data/jpa/DataJPATestCheckpoint.java:84]: Done

File: dev/io.openliberty.data.internal_fat/fat/src/test/jakarta/data/DataTestCheckpoint.java (hunk 2 above)
Comment (tjwatson) [reply dev/io.openliberty.data.internal_fat/fat/src/test/jakarta/data/DataTestCheckpoint.java:100]: Done

**Review by KyleAure — APPROVED** [review]

**Comment by tjwatson** [comment]
#libby

**Pull request merged**